### PR TITLE
Fix Platform.localeName by setting _Platform._localeClosure

### PR DESCRIPTION
**Pull request by zanderso**
For https://github.com/flutter/flutter/issues/13363

### CODE REVIEW

File: lib/ui/hooks.dart [hunk 1]
```diff
@@ -39,6 +39,10 @@ void _updateWindowMetrics(double devicePixelRatio,
   _invoke(window.onMetricsChanged, window._onMetricsChangedZone);
 }
 
+String _localeClosure() => window._locale.toString();
+
+Function _getLocaleClosure() => _localeClosure;
```
Comment (Hixie) [inline lib/ui/hooks.dart]: would it make sense to use a typedef?

Comment (zanderso) [reply lib/ui/hooks.dart]: Done.

**Comment by aam** [comment]
@zanderso have you tried running flutter gallery with this engine change?
As @crelier is trying to do daily dart roll he is hitting issue running flutter gallery in release mode: https://github.com/flutter/engine/pull/4343#issuecomment-351468993

**Comment by zanderso** [comment]
Yes. There is a bug in flutter_tool here: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/commands/build_aot.dart#L196 in which it fails to use the updated libraries from the local engine. A temporary workaround is to manually copy //engine/src/flutter/lib/ui/hooks.dart to //flutter/bin/cache/pkg/sky_engine/lib/ui/hooks.dart and //flutter/bin/cache/pkg/sky_engine/sdk_ext/hooks.dart.

**Comment by Hixie** [comment]
When running into a bug, please feel empowered to fix the bug rather than just work around it. :-)

**Pull request merged**